pkg/scalers/honeycomb_scaler.go

```diff
@@ -0,0 +1,272 @@
+	req, _ := http.NewRequestWithContext(ctx, "POST", createURL, bytes.NewBuffer(bodyBytes))
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Honeycomb-Team", s.metadata.APIKey)
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("honeycomb create query error: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		body, _ := io.ReadAll(resp.Body)
+		return 0, fmt.Errorf("honeycomb createQuery status: %s - %s", resp.Status, string(body))
+	}
+	var createRes struct{ ID string `json:"id"` }
+	if err := json.NewDecoder(resp.Body).Decode(&createRes); err != nil {
+		return 0, fmt.Errorf("decode createQuery: %w", err)
+	}
+	if createRes.ID == "" {
+		return 0, errors.New("createQuery: missing query id")
+	}
+
+	// 2. Run Query
+	runURL := fmt.Sprintf("%s/query_results/%s", honeycombBaseURL, s.metadata.Dataset)
+	runBody, _ := json.Marshal(map[string]interface{}{
+		"query_id":                   createRes.ID,
+		"disable_series":             false,
+		"disable_total_by_aggregate": true,
+		"disable_other_by_aggregate": true,
+		"limit":                      10000,
+	})
+	runReq, _ := http.NewRequestWithContext(ctx, "POST", runURL, bytes.NewBuffer(runBody))
+	runReq.Header.Set("Content-Type", "application/json")
+	runReq.Header.Set("X-Honeycomb-Team", s.metadata.APIKey)
+	runResp, err := s.httpClient.Do(runReq)
+	if err != nil {
+		return 0, fmt.Errorf("honeycomb run query error: %w", err)
+	}
+	defer runResp.Body.Close()
+	if runResp.StatusCode == 429 {
+		return 0, errors.New("honeycomb: rate limited (429), back off and try again later")
+	}
+	if runResp.StatusCode != 200 && runResp.StatusCode != 201 {
+		body, _ := io.ReadAll(runResp.Body)
+		return 0, fmt.Errorf("honeycomb runQuery status: %s - %s", runResp.Status, string(body))
+	}
+	var runRes struct {
+		ID       string `json:"id"`
+		Complete bool   `json:"complete"`
+		Data     struct {
+			Results []map[string]interface{} `json:"results"`
+		} `json:"data"`
+	}
+	if err := json.NewDecoder(runResp.Body).Decode(&runRes); err != nil {
+		return 0, fmt.Errorf("decode runQuery: %w", err)
+	}
+	if runRes.ID == "" {
+		return 0, errors.New("runQuery: missing queryResult id")
+	}
+	if runRes.Complete && len(runRes.Data.Results) > 0 {
+		return extractResultField(runRes.Data.Results, s.metadata.ResultField)
+	}
+
+	// 3. Poll for completion (exponential backoff)
+	pollURL := fmt.Sprintf("%s/query_results/%s/%s", honeycombBaseURL, s.metadata.Dataset, runRes.ID)
+	pollDelay := initialPollDelay
+	for attempt := 0; attempt < maxPollAttempts; attempt++ {
+		time.Sleep(pollDelay)
+		pollDelay *= 2
+		statusReq, _ := http.NewRequestWithContext(ctx, "GET", pollURL, nil)
+		statusReq.Header.Set("X-Honeycomb-Team", s.metadata.APIKey)
+		statusResp, err := s.httpClient.Do(statusReq)
+		if err != nil {
+			return 0, fmt.Errorf("honeycomb poll query error: %w", err)
+		}
+		defer statusResp.Body.Close()
```

Comment:
I dont thin this is a good idea as it might lead to resource leaks as these defer called will be accumulated inside the polling loop. 
Do you think we should `statusResp.Body.Close()` at the end of the loop ?